### PR TITLE
Add types/uuencode

### PR DESCRIPTION
**Pull request by synaestheory**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/uuencode/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for uuencode 0.0
+// Project: https://github.com/zacbarton/node-uuencode#readme
+// Definitions by: Seth Niimi <https://github.com/me>
```
Comment (danger-public) [inline types/uuencode/index.d.ts:3]: - :no_entry_sign: This line should have your github username in it, not /me

Comment (synaestheory) [reply types/uuencode/index.d.ts]: Updated to use synaesthoery in both places.

File: types/uuencode/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for uuencode 0.0
+// Project: https://github.com/zacbarton/node-uuencode#readme
+// Definitions by: synaestheory <https://github.com/synaesthoery>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function decode(str: string): string;
+
+export function encode(str: string): string;
```
Comment (DanielRosenwasser) [inline types/uuencode/index.d.ts:8]:
```suggestion
export function encode(str: string | Buffer): string;
```

File: types/uuencode/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for uuencode 0.0
+// Project: https://github.com/zacbarton/node-uuencode#readme
+// Definitions by: synaestheory <https://github.com/synaesthoery>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function decode(str: string): string;
```
Comment (DanielRosenwasser) [inline types/uuencode/index.d.ts:6]:
```suggestion
export function decode(str: string | Buffer): string;
```

**Review by DanielRosenwasser — COMMENTED** [review]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either [make sure your GitHub account is associated with the email address](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/adding-an-email-address-to-your-github-account) that you're using for your commits, or [rebase and amend your commits](https://stackoverflow.com/a/31669789) to fix the author name and email.

If you don't care about any of this stuff, I can always just pull in the changes.

Also, the source seems to imply these can take Buffers.

**Comment by synaestheory** [comment]
@DanielRosenwasser All updated. I use gitlab primarily and hadn't realized that I missed updating github to also include the newer e-mail.  Thanks for catching that.

**Pull request merged**